web/src/components/organisms/Settings/Members/hooks.ts

```diff
@@ -100,30 +109,28 @@ export default ({ workspaceId }: Props) => {
       .sort((user1, user2) => stringSortCallback(user1.userId, user2.userId));
   }, [currentWorkspace, searchTerm]);
 
-  const [addUserToWorkspaceMutation] = useAddUserToWorkspaceMutation();
-
-  const handleMemberAddToWorkspace = useCallback(
-    async (userIds: string[]) => {
-      const results = await Promise.all(
-        userIds.map(async userId => {
-          if (!workspaceId) return;
-          const result = await addUserToWorkspaceMutation({
-            variables: { userId, workspaceId, role: Role.Reader },
-            refetchQueries: ["GetWorkspaces"],
-          });
-          const workspace = result.data?.addUserToWorkspace?.workspace;
-          if (result.errors || !workspace) {
-            Notification.error({ message: t("Failed to add one or more members.") });
-            return;
-          }
-          setWorkspace(workspace);
-        }),
-      );
-      if (results) {
-        Notification.success({ message: t("Successfully added member(s) to the workspace!") });
-      }
-    },
-    [workspaceId, addUserToWorkspaceMutation, setWorkspace, t],
+  const [addUsersToWorkspaceMutation] = useAddUsersToWorkspaceMutation();
+
+  const handleUsersAddToWorkspace = useCallback(
+    (users: MemberInput[]) =>
+      (async () => {
+        if (!workspaceId) return;
+        const result = await addUsersToWorkspaceMutation({
+          variables: { workspaceId, users: users as GQLMemberInput[] },
+          refetchQueries: ["GetWorkspaces"],
+        });
+        const workspace = result.data?.addUsersToWorkspace?.workspace;
+        if (result.errors || !workspace) {
+          Notification.error({ message: t("Failed to add one or more members.") });
+          return;
+        }
+        setWorkspace(workspace);
+
+        if (result) {
```

Comment:
I think you can check like `if (result.workspace)` or something to be a little more specific, no?

Reply:
How about `if (result.data)`?

Reply:
That! haha Because if my memory is correct if there is an error there won't be result.data.

Reply:
So which one should it be?
result.data
result.context
result.extensions

Reply:
How you updated it is good I think (result.data)!